src/hotspot/share/runtime/arguments.cpp

```diff
@@ -3586,7 +3586,7 @@ static bool use_vm_log() {
   if (LogCompilation || !FLAG_IS_DEFAULT(LogFile) ||
       PrintCompilation || PrintInlining || PrintDependencies || PrintNativeNMethods ||
       PrintDebugInfo || PrintRelocations || PrintNMethods || PrintExceptionHandlers ||
-      PrintAssembly || TraceDeoptimization || TraceDependencies ||
+      PrintAssembly || TraceDeoptimization || log_is_enabled(Debug, dependencies) ||
```

Comment:
Now TraceDependencies is converted to UL I think it should just be deleted from this function. We don't need to enable LogVMOutput in that case.

Reply:
Sorry I don't follow that. `use_vm_log()` only affects non-product builds and forces `LogVMOutput` to true. That in turn will cause `defaultStream::init_log()` to execute which initializes the log file etc. But I don't see how that would cause UL logging for "dependencies" to also get written to the log file???

Reply:
Hmmm this seems broken to me then. We output the general dependency logging to one place based on UL configuration, but then we output `log_dependency` to the log file. If these are meant to be related and always reported together then we have lost that. If they are actually unrelated then this may still need the TraceDependencies flag to control it.

@iwanowww can you comment on this please?

Reply:
Maybe we don't need this -Xlog test in this place, because this is also what PrintDependencies does.

Reply:
Yes, UL check doesn't help here. Irrespective of the check, UL output isn't automatically placed in VM log.
It's convenient for LogCompilation to have VM output along with other VM events present in the log. 
Having said that, I'm fine with dropping TraceDependencies check here and improve the interaction between UL and LogVMOutput separately.

Reply:
Okay thank you.